### PR TITLE
Reformat `formulas.sls` Jinja to make it easier to work with

### PR DESCRIPTION
**Pull request by myii**
NOTE: This refactor does nothing more than modify whitespace.  No change is made to the functionality at all.

---

While looking at #403, the logic is incredibly difficult to follow.  This PR adapts the Jinja to make each much easier to comprehend and work with.

---

It's easier to appreciate the changes by viewing the [whole file](https://github.com/saltstack-formulas/salt-formula/blob/f76364c1b7afffc435656a76b8c3ed49b364579d/salt/formulas.sls) rather than the diff.

Update: After the second commit, this is the [whole file](https://github.com/saltstack-formulas/salt-formula/blob/922135df31875e928e75dc5ad18dcaba7cb3ae0f/salt/formulas.sls).

### CODE REVIEW

**Comment by myii** [comment]
@getSurreal Really appreciate you taking the time to review.  You make some excellent points.  Let me explain the method to my madness:

* Readability/comprehension:
    * It seems we're both agreeing that using some kind of indent helps.
    * Some formulas indent the whole statement and I still prefer that in some places, such as blocks of Jinja such as macros.  Otherwise, there are other considerations, as mentioned below.
* Indent/folding:
    * The basic idea of keeping all Jinja flush to the LHS makes a lot of sense.
    * However, this can seriously interfere with readability with the surrounding code -- and the Jinja is there to supplement, not to take over!
    * So keeping the Jinja aligned with the surrounding code is a practice that many have adopted.
    * Furthermore, this is really important when using code folding in your choice of editor -- otherwise, your folds get broken at every Jinja statement.

The rest of the points I'll make inline.  Thanks again.

File: salt/formulas.sls [hunk 1]
```diff
@@ -1,68 +1,70 @@
...
+{%-       do processed_basedirs.append(basedir) %}
 {{ basedir }}:
   file.directory:
-    {%- for key, value in salt['pillar.get']('salt_formulas:basedir_opts',
-                                             {'makedirs': True}).items() %}
+    {%-   for key, value in salt['pillar.get'](
+            'salt_formulas:basedir_opts',
+            {'makedirs': True}
+          ).items() %}
     - {{ key }}: {{ value }}
```
Comment (myii) [inline salt/formulas.sls]: @getSurreal I accept your point here.  The fallback here is the line continuation style that you're used to in Python.  You can see mine!  It's a very opinionated topic but I'm fairly relaxed about this and happy to reset here, except for the slight indent of the `for` to match the surrounding Jinja.

**Comment by aboe76** [comment]
@myii and @getSurreal can this be merged?

**Comment by myii** [comment]
@aboe76 Thanks for the review.  I'm finished here but I'll let @getSurreal have the last word.

**Comment by getSurreal** [comment]
@myii I'm just not following why sometimes you indent after {%- and sometimes before.  However, I'm not blocking the merge if no one else minds that.

**Comment by aboe76** [comment]
@myii thanks for cleaning up this file.

**Comment by myii** [comment]
@getSurreal Perhaps this will help -- pretty much everything stripped out except the Jinja:

```jinja
{%- set processed_gitdirs = {} %}
{%- set processed_gitdir_envs = [] %}
{%- set processed_basedirs = [] %}
{%- from "salt/map.jinja" import formulas_settings with context %}
{%- from "salt/formulas.jinja" import formulas_git_opt with context %}
{%- from "salt/formulas.jinja" import formulas_opts_for_git_latest with context %}
{%- for env, entries in salt['pillar.get']('salt_formulas:list', {}).items() %}
{%-   for entry in entries %}
{%-     set basedir = formulas_git_opt(env, 'basedir')|load_yaml %}
{%-     set gitdir = '{0}/{1}'.format(basedir, entry) %}
{%-     set update = formulas_git_opt(env, 'update')|load_yaml %}
{%-     if formulas_settings.checkout_orig_branch %}
{%-       if not salt['file.directory_exists']('{0}/{1}'.format(gitdir, '.git')) %}
{%-         set gitdir_branch = '' %}
{%-       else %}
{%-         set gitdir_branch = salt['git.current_branch'](gitdir) %}
{%-       endif %}
{%-       do processed_gitdirs.update({gitdir:gitdir_branch}) %}
{%-     endif %}
{%-     if basedir not in processed_basedirs %}
{%-       do processed_basedirs.append(basedir) %}
    {%-   for key, value in salt['pillar.get']('salt_formulas:basedir_opts',
    - {{ key }}: {{ value }}
    {%-   endfor %}
{%-     endif %}
{%-     set gitdir_env = '{0}_{1}'.format(gitdir, env) %}
{%-     if gitdir_env not in processed_gitdir_envs %}
{%-       do processed_gitdir_envs.append(gitdir_env) %}
{%-       set options = formulas_opts_for_git_latest(env)|load_yaml %}
{%-       set baseurl = formulas_git_opt(env, 'baseurl')|load_yaml %}
    {%-   for key, value in options.items() %}
    - {{ key }}: {{ value }}
    {%-   endfor %}
    {%-   if not update %}
    - onlyif: rm -fr {{ gitdir }} >/dev/null 2>&1 | true
    {%-   endif %}
{%-     endif %}
{%-   endfor %}
{%- endfor %}
{%- if formulas_settings.checkout_orig_branch %}
{%-   for gitdir, original_branch in processed_gitdirs.items() %}
{%-     if original_branch %}
{%-       set gitdir_user = salt['file.get_user'](gitdir) %}
{%-     endif %}
{%-   endfor %}
{%- endif %}
```

So pretty much Python indentation throughout.  The start of the `{%` aligned with the surrounding `sls` code.

**Comment by myii** [comment]
@aboe76 Thanks for the merge.

**Comment by aboe76** [comment]
@myii does your new proposal render a valid saltstack state sls?

**Comment by myii** [comment]
@aboe Which proposal?  The one in the comment above?  That's just to show only the Jinja -- to show why I've indented it that way I have.

**Comment by aboe76** [comment]
@myii ok,

**Pull request merged**